cppwinrt/component_writers.h

```diff
@@ -988,9 +988,22 @@ namespace winrt::@::implementation
     static void write_generated_static_assert(writer& w)
     {
         auto format = R"(
-// Note: Remove this static_assert after copying these generated source files to your project.
-// This assertion exists to avoid compiling these generated source files directly.
-static_assert(false, "Do not compile generated C++/WinRT source files directly");
+// WARNING: This file is automatically generated from IDL. Do not directly
+// add this file to your project, as any changes you make will be lost.
```

Comment:
Perhaps also

```cpp
// This file is a stub you can use as a starting point for your implementation.
```

Reply:
that is the most important point, the rest is the machinery that support it.